src/main/java/com/alphawallet/attestation/demo/Demo.java

```diff
@@ -227,11 +227,11 @@ private static void receiveCheque(Path pathUserKey, Path chequeSecretDir,
     }
     // TODO how should this actually be?
     SmartContract sc = new SmartContract();
-    /*FullProofOfExponent newPok = new FullProofOfExponent(redeem.getDerEncoding());
-    if (!sc.testEncoding(redeem.getPok())) {
+    FullProofOfExponent newPok = new FullProofOfExponent(redeem.getDerEncoding());
```

Comment:
I think we might have misunderstood each other. The redeem request only contains a UsageProofOfKnowledge, which is what should be given as argument to the smart contract along with the Ticket and Attestation commitments. 
What is happening now is that you are trying to construct a FullProofOfExponent from a UseDevconTicket object. This fails with an exception, but becasuse the Demo class actually handles all exceptions this is not seen by the JUnit test.
This is a mistake of mine; I didn't verify that the test would fail when an internal method fails. I have committed a small change to Demo on this branch to ensure that the test fails when something goes wrong.

Reply:
OK, I didn't check what was going on there; since the test doesn't work from gradle the way it is set up.

Reply:
Fixed. I actually was editing the wrong contract, confusion as the file positions and names keep jumping around all over the repo.